### PR TITLE
Pull in some object store changes

### PR DESCRIPTION
**Pull request by bdash**
These changes are primarily recent work related to formatting error messages. The aim here is simply to reduce the diff between our version of the object store and upstream to avoid future merge issues.

### CODE REVIEW

**Comment by austinzheng** [comment]
Looks good to me.

File: Realm/ObjectStore/object_store.cpp [hunk 1]
```diff
@@ -725,60 +736,75 @@ InvalidNullabilityException::InvalidNullabilityException(std::string const& obje
...
-MismatchedPropertiesException::MismatchedPropertiesException(std::string const& object_type, Property const& old_property, Property const& new_property) :
+MismatchedPropertiesException::MismatchedPropertiesException(std::string const& object_type,
+                                                             Property const& old_property,
+                                                             Property const& new_property) :
     ObjectSchemaValidationException(object_type), m_old_property(old_property), m_new_property(new_property)
 {
     if (new_property.type != old_property.type) {
-        m_what = "Property types for '" + old_property.name + "' property do not match. Old type '" + string_for_property_type(old_property.type) +
-        "', new type '" + string_for_property_type(new_property.type) + "'";
+        m_what = util::format("Property types for '%1' property doe not match. Old type '%2', new type '%3'.",
```
Comment (austinzheng) [inline Realm/ObjectStore/object_store.cpp]: "does not"

**Comment by austinzheng** [comment]
It still looks good.

**Comment by tgoyne** [comment]
👍

**Pull request merged**